x-pack/plugins/upgrade_assistant/public/application/components/overview/migrate_system_indices/migrate_system_indices.tsx

```diff
@@ -123,10 +124,55 @@ const i18nTexts = {
   },
 };
 
+const ConfirmModal: React.FC<{
+  onCancel: () => void;
+  onConfirm: () => void;
+}> = ({ onCancel, onConfirm }) => (
+  <EuiConfirmModal
+    title={i18n.translate('xpack.upgradeAssistant.overview.systemIndices.confirmModal.title', {
+      defaultMessage: 'Migrate Indices',
+    })}
+    onCancel={onCancel}
+    onConfirm={onConfirm}
+    cancelButtonText={i18n.translate(
+      'xpack.upgradeAssistant.overview.systemIndices.confirmModal.cancelButton.label',
+      {
+        defaultMessage: 'Cancel',
+      }
+    )}
+    confirmButtonText={i18n.translate(
+      'xpack.upgradeAssistant.overview.systemIndices.confirmModa.confirmButton.label',
+      {
+        defaultMessage: 'Confirm',
+      }
+    )}
+    defaultFocusedButton="confirm"
+  >
+    {i18n.translate('xpack.upgradeAssistant.overview.systemIndices.confirmModa.description', {
+      defaultMessage:
+        'Migrating system indices may lead to downtime while the indices are reindexed. Are you sure you want to migrate system indices?',
+    })}
+  </EuiConfirmModal>
+);
+
 const MigrateSystemIndicesStep: FunctionComponent<Props> = ({ setIsComplete }) => {
   const { beginSystemIndicesMigration, startMigrationStatus, migrationStatus, setShowFlyout } =
     useMigrateSystemIndices();
 
+  const [isModalVisible, setIsModalVisible] = useState(false);
+
+  const refreshButtonClick = () => {
```

Comment:
how do you feel about something slightly more descriptive such as `openMigrationModal`?

Reply:
`openMigrationModal` is more semantic, I have updated it